lib/spack/spack/test/database.py

```diff
@@ -161,15 +161,35 @@ def test_default_queries(database):
 
     libraries = spec['zmpi'].libs
     assert len(libraries) == 1
+    assert libraries.names[0] == 'zmpi'
 
     headers = spec['zmpi'].headers
     assert len(headers) == 1
+    assert headers.names[0] == 'zmpi'
 
     command = spec['zmpi'].command
     assert isinstance(command, Executable)
     assert command.name == 'zmpi'
     assert os.path.exists(command.path)
 
+    install_db = database.mock.db
+    rec = install_db.get_record('libelf')
+
+    spec = rec.spec
+
+    libraries = spec['libelf'].libs
+    assert len(libraries) == 1
+    assert libraries.names[0] == 'elf'
```

Comment:
Without these changes, this would return `libelf`, meaning the library it picks up is called `liblibelf.so`.

Reply:
Can you add a comment here saying something like:
```python
# Testing a package with a name starting with 'lib', to ensure
# the library doesn't have a double 'lib' prefix
```

Reply:
Done